### PR TITLE
Feature/hooks: Turn class components into functional components

### PR DESCRIPTION
**Pull request by nikoescobal**
In this pull request, I did the following:
- Convert all class components into functional components
- Utilize React syntax and best practices
- Refactor code to keep it clean

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,239 +1,236 @@
...
-    };
-  }
+const Calculator = () => {
+  const operators = '+-x%÷';
+  const numbers = '0123456789';
+  const [total, setTotal] = useState('');
+  const [next, setNext] = useState('');
+  const [operation, setOperation] = useState('');
+  const [expression, setExpression] = useState('0');
+  const [completed, setCompleted] = useState(false);
```
Comment (dasileker) [inline src/components/Calculator.js:12]: - [ ] The hooks are meant to be for calculating not to refactor all, so what you did here you refactored the buttons to be hooked but the requirement says you need to refactor the base class to a function that will take a hook.

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @nikoescobal,** 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

- I  can't do any operation after you refactor your code, please take a look at the video below.


CPT2110301244-642x305




## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by nikoescobal** [comment]
@dasileker I've implemented the requested changes. Hope this gets approved soon.

File: src/components/Calculator.js [hunk 2]
```diff
@@ -1,239 +1,236 @@
...
+        setTotal(event.target.name);
+        setExpression(event.target.name);
       } else {
-        this.setState({ next: event.target.name });
-        const exp = this.state.expression;
-        this.setState({
-          expression: `${exp} ${event.target.name}`,
-        });
+        setNext(event.target.name);
+        // const exp = expression;
```
Comment (cyonii) [inline src/components/Calculator.js:50]: - [ ] [ OPTIONAL ] Please do not leave commented code in your codebase. It makes your work messy and unprofessional. Use git history if you need to review a different version of your code.

**Review by cyonii — APPROVED** [review]
Hi @nikoescobal, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by nikoescobal** [comment]
Thank you @cyonii I appreciate it :) Now I can sleep haha

**Pull request merged**